src/fmu/dataio/providers/_filedata.py

```diff
@@ -104,6 +104,15 @@ def _add_filename_to_path(self, path: Path) -> Path:
         stem = self._get_filestem()
         return (path / stem).with_suffix(path.suffix + self.objdata.extension)
 
+    def _parent_name_handling(self) -> str:
+        """Action when both parent key and geometry key are given (GridProperty)."""
+        if not self.objdata.get_geometry():
+            return self.dataio.parent
+
+        geom = self.objdata.get_geometry()
+
+        return geom.name if geom else ""
```

Comment:
To align with how we treat the `name` which we set as a property, we could do the same for parent here.
```suggestion
    @property
    def parent(self) -> str:
        """Action when both parent key and geometry key are given (GridProperty)."""
        geom = self.objdata.get_geometry()
        return geom.name if geom else self.dataio.parent
```

Reply:
Also should we issue a warning to say that the 'parent' argument is ignored? to avoid confusion 🙂